GA4 formatting fix (#548)
rudderlabs/rudderstack-docs: destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode.mdx

# Cloud Mode

> RudderStack lets you send your event data to Google Analytics 4 destination via the cloud mode.

Reviewer: ```suggestion
RudderStack lets you send your event data to Google Analytics 4 via the cloud mode.
```

Reviewer: Should we also add the link to the connection modes guide, specifically the cloud mode section in it?

Author: Its already there, just below this line

Reviewer: oh sorry - think it got hidden

The [`track`](https://www.rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-spec/track/) call lets you capture user events along with the properties associated with them.

> RudderStack supports both the `gtag` and `firebase` ways for tagging in websites in the cloud mode. However:

Reviewer: Do we have any supporting Google Ads doc references for these methods? We should link them for more clarity.

Reviewer 2: The supporting doc - https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference?client_type=gtag#payload

Reviewer: @ishachopra0401 let's incorporate this for better clarity.

- If you use `gtag`, passing the `client_id` parameter becomes mandatory.
- If you use `firebase`, passing the `app_instance_id` parameter becomes mandatory.

The mappings for the above parameters are listed below:

| Parameters    | Mapping value  |
| :-------------| :-----------------|
| client_id         |  `context.client_id`  or `messageId` |
| app_instance_id   |  from externalID `ga4AppInstanceId`  |

> A sample `track` call when `gtag` is used, is as shown:

Reviewer: ```suggestion
A sample `track` call when `gtag` is used is as shown:
```

> A sample `track` call where `gtag` is used, is as shown:

Reviewer 2: ```suggestion
A sample `track` call when `gtag` is used, is as shown:
```

```javascript

> rudderanalytics.track('product list viewed', {

Reviewer: ```suggestion
rudderanalytics.track('Product List Viewed', {
```

Reviewer: Please confirm this change, @akashrpo - as it's a standard e-Commerce event, I believe the event name must be written as `Product List Viewed` instead of `product list viewed` (lower case).

Reviewer 2: We can add this as we are anyway bringing the strings to a common ground by lowercasing the names that we receive.

    item_list_name: "Related Products",
    location_id: "L_12345"
  }]
}, {
  client_id: "client_id"
});
```

> A sample `track` call when `firebase` is used, is as shown:

Reviewer: ```suggestion
A sample `track` call when `firebase` is used is shown below:
```

```javascript

> rudderanalytics.track('product list viewed', {

Reviewer: Same comment as above, this should be 'Product List Viewed'.

  list_id: "related_products",
  category: "Related_products",
  products: [{
    product_id: "507f1f77bcf86cd799439011",
    name: "Monopoly: 3rd Edition",
    coupon: "SUMMER_FUN",
    category: "Apparel",
    brand: "Google",
    variant: "green",
    price: "19",
    quantity: "2",
    position: "1",
    affiliation: "Google Merchandise Store",
    currency: "USD",
    discount: 2.22,
    item_category2: "Adult",
    item_category3: "Shirts",
    item_category4: "Crew",
    item_category5: "Short sleeve",
    item_list_id: "related_products",
    item_list_name: "Related Products",
    location_id: "L_12345"
  }]
}, {
  externalId: [{
    type: "ga4AppInstanceId",
    id: "f0dd99b6f979fb551ce583373900f937"
  }],
});
```

## E-commerce

> RudderStack supports e-commerce tracking for Google Analytics 4. You can use the [RudderStack e-commerce specification](https://rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-ecommerce-events-specification/) for sending events while instrumenting your site with the RudderStack SDK.

Reviewer: ```suggestion
RudderStack supports e-commerce tracking for Google Analytics 4. You can refer to the [RudderStack E-commerce Specification](https://rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-ecommerce-events-specification/) guide for sending events while instrumenting your site with the RudderStack SDK.
```

> The below table lists the RudderStack and Google Analytics 4 events mappings:

Reviewer: ```suggestion
The following table lists the RudderStack and Google Analytics 4 events mappings:
```

  </tr>
   <tr>
    <td>Checkout Started</td>
    <td><code class="inline-code">begin_checkout</code></td>
  </tr>
   <tr>
    <td>Order Completed<br/>Order Refunded</td>
    <td><code class="inline-code">purchase</code><br /><code class="inline-code">refund</code></td>
  </tr>
    <tr>
    <td>Product Shared<br/>Cart Shared</td>
    <td><code class="inline-code">share</code></td>
  </tr>
   <tr>
    <td>Group</td>
    <td><code class="inline-code">join_group</code></td>
  </tr>
  <tr>
    <td>Payment Info Entered</td>
    <td><code class="inline-code">add_payment_info</code><br/>OR<br/><code class="inline-code">add_shipping_info</code></td>
  </tr>
</table>

<div class="infoBlock">

The mapping of the <strong>Payment Info Entered</strong> event to the <code class="inline-code">add_payment_info</code> or <code class="inline-code">add_shipping_info</code> event is determined on the basis of parameters passed to it as explained <a href="https://www.rudderstack.com/docs/destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode/#:~:text=currency-,The%20Payment%20Info%20Entered%20event%20is%20mapped%20on%20the%20basis,event.,-Most%20of%20the">here</a>.
</div>

> The below table lists the RudderStack and Google Analytics 4 properties mappings based on the specific RudderStack events:

Reviewer: ```suggestion
The following table lists the RudderStack and Google Analytics 4 properties mappings based on the specific RudderStack events:
```

| Product Added to Wishlist           | Required  |
| View Search Results           | Optional  |

> The below table details out the mappings for above events for the `Items` array:

Reviewer: ```suggestion
The below table details the parameter mappings for the above events in the `Items` array:
```

| Promotion Viewed          | Required        |
| Promotion Clicked      | Optional     |

> The below table details out the mappings for above events for the `Items` array:

Reviewer: ```suggestion
The following table details the parameter mappings present in the `Items` array, for the above events:
```

  </tr>
</table>

<div class="infoBlock">
  

> You can pass the custom user properties to any of the events by passing it as `properties.user_property`. Refer to the <a href="https://developers.google.com/analytics/devguides/collection/protocol/ga4/user-properties?client_type=gtag">User properties</a> for more information.

Reviewer 2: ```suggestion
You can pass the custom user properties to any of the events by passing it as `properties.user_properties`. Refer to the <a href="https://developers.google.com/analytics/devguides/collection/protocol/ga4/user-properties?client_type=gtag">User properties</a> for more information.
```

> You can pass the custom user properties to any of the events by passing it as `properties.user_properties`. Refer to the <a href="https://developers.google.com/analytics/devguides/collection/protocol/ga4/user-properties?client_type=gtag">User properties</a> for more information.

Reviewer: ```suggestion
You can pass the custom user properties to any of the events by passing them as `properties.user_properties`. Refer to the <a href="https://developers.google.com/analytics/devguides/collection/protocol/ga4/user-properties?client_type=gtag">Google Analytics 4 documentation</a> for more information.
```
